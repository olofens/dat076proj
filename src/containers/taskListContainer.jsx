import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from "redux";
import { showModal, hideModal, startEditTask, finishEditTask } from "../actions/index.js"

import TaskListComponent from '../components/taskListComponent.jsx';
import Droppable from "./droppable.jsx"

class TaskList extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            
                <TaskListComponent
                tasks={this.props.tasks}
                openEdit={this.props.startEditTask}
                closeEdit={this.props.finishEditTask}
                showModal={this.props.showModal}
                hideModal={this.props.hideModal}
                modalIsOpen={this.props.modalIsOpen}

            />
        );
    }
}

function mapStateToProps(state) {
    return {
        tasks: state.todoTasks,
        modalIsOpen: state.modalIsOpen
    }
}

function matchDispatchToProps(dispatch) {
    return bindActionCreators({ 
        startEditTask: startEditTask,
        finishEditTask: finishEditTask,
        showModal: showModal,
        hideModal: hideModal }, dispatch)
}

export default connect(mapStateToProps, matchDispatchToProps)(TaskList);