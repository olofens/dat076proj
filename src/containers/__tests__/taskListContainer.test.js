import React from 'react';
import { shallow } from 'enzyme';
import taskListContainer from '../taskListContainer.jsx';


describe('taskListContainer', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<taskListContainer debug />);
  
    expect(component).toMatchSnapshot();
  });
});