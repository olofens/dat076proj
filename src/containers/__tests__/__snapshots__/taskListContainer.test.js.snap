// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`taskListContainer should render correctly in "debug" mode 1`] = `
<taskListContainer
  debug={true}
/>
`;
